### PR TITLE
iptables

### PR DESCRIPTION
**Pull request by baude**
implement iptables rules for NAT and MASQ setup.  At this time, no tear
down is being proposed.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: src/commands/setup.rs [hunk 1]
```diff
@@ -29,7 +33,66 @@ impl Setup {
...
+                self.setup_iptables(&network_name, nets);
+            }
+        }
         debug!("{:?}", "Setup complete");
-        //()
+    }
+    /// Set up required iptables for NAT
+    ///
+    ///
+    pub fn setup_iptables(&self, network_name: &str, subnets: Vec<Subnet>) {
```
Comment (flouthoc) [inline src/commands/setup.rs:49]: I think can have this function in `network` create not at the frontend.

File: src/commands/setup.rs [hunk 2]
```diff
@@ -29,7 +33,66 @@ impl Setup {
...
+        debug!("iptables");
+        for net in subnets{
+            // Basic object setup
+            let ipt = iptables::new(false).unwrap();
```
Comment (flouthoc) [inline src/commands/setup.rs:53]: not-a-blocker: unwrap from serializers might be okay but for iptables we might wanna handle error. Can revisit this later.

File: src/commands/setup.rs (hunk 1 above)
Comment (baude) [reply src/commands/setup.rs:49]: can you be more exact. point to exactly where

File: src/commands/setup.rs [hunk 3]
```diff
@@ -29,7 +33,66 @@ impl Setup {
...
+            //  Check if the chain exists, if not - create it
+            let chain_check = ipt.chain_exists(NAT, network_name);
+            match chain_check {
+                Ok(true) => debug!("{} chain exists", &network_name),
+                Ok(false) => {
+                    // The chain did not exist
+                    debug!("need to create chain {}", network_name);
+                    match ipt.new_chain(NAT, network_name) {
+                        Ok(_) => debug!("chain {} created", network_name),
+                        Err(e) => panic!("{}", e)
```
Comment (flouthoc) [inline src/commands/setup.rs:63]: We can return CliError instead of panic. Can revisit this later.

File: src/network/types.rs [hunk 4]
```diff
@@ -134,14 +135,14 @@ pub struct NetAddress {
...
 // TODO: [Not important] Transformed from golang net.IPNet. Swtich to native rust if needed.
-#[derive(Debug, Serialize, Deserialize)]
-pub struct IpNet {
-    #[serde(rename = "IP")]
-    pub ip: String,
-
-    #[serde(rename = "Mask")]
-    pub mask: String,
-}
+// #[derive(Debug, Serialize, Deserialize)]
```
Comment (flouthoc) [inline src/network/types.rs:138]: We can remove this if its not needed at all.

File: src/commands/setup.rs (hunk 1 above)
Comment (flouthoc) [reply src/commands/setup.rs:49]: @baude we could create a new file `src/network/core.rs` maybe like this ->  https://github.com/containers/netavark/pull/17

**Pull request merged**